homeassistant/components/imap_email_content/sensor.py

```diff
@@ -95,9 +95,22 @@ def __init__(self, user, password, server, port, folder, verify_ssl):
         self._folder = folder
         self._verify_ssl = verify_ssl
         self._last_id = None
+        self._last_message = None
         self._unread_ids = deque([])
         self.connection = None
 
+    @property
+    def last_id(self) -> int | None:
+        """Return last email uid that was processed."""
+        return self._last_id
+
+    @property
+    def last_unread_id(self) -> int | None:
+        """Return last email uid received."""
+        if self._unread_ids:
+            return max(int(uid) for uid in self._unread_ids)
```

Comment:
I think this assumption that they will always increase is safe 
https://www.rfc-editor.org/rfc/rfc3501#section-2.3.1.1

Reply:
But will the list be alphabetical, and always have the same order? But yes, that is what I have seen, the UID seems to be an integer (just need to convert it).

Reply:
It might be safer to use a binary comparison instead of converting to an int

Reply:
We coud use `int(uid_str, base=16)`

Reply:
I'm still looking for the RFC described way to sort these...
found https://www.rfc-editor.org/rfc/rfc5256.html but probably doesn't help

Reply:
https://stackoverflow.com/questions/12409862/python-imap-the-order-of-uids